tests/components/history_stats/test_config_flow.py

```diff
@@ -193,3 +199,350 @@ async def test_entry_already_exist(
+                State(
+                    monitored_entity,
+                    "on",
+                    last_changed=start_time,
+                    last_updated=start_time,
+                ),
+                State(
+                    monitored_entity,
+                    "off",
+                    last_changed=t1,
+                    last_updated=t1,
+                ),
+                State(
+                    monitored_entity,
+                    "on",
+                    last_changed=t2,
+                    last_updated=t2,
+                ),
+            ]
+        }
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] is None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_NAME: DEFAULT_NAME,
+            CONF_ENTITY_ID: monitored_entity,
+            CONF_TYPE: CONF_TYPE_COUNT,
+            CONF_STATE: ["on"],
+        },
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "options"
+    assert result["errors"] is None
+    assert result["preview"] == "history_stats"
+
+    with (
+        patch(
+            "homeassistant.components.recorder.history.state_changes_during_period",
+            _fake_states,
+        ),
+        freeze_time(t3),
+    ):
+        await client.send_json_auto_id(
+            {
+                "type": "history_stats/start_preview",
+                "flow_id": result["flow_id"],
+                "flow_type": "config_flow",
+                "user_input": {
+                    CONF_ENTITY_ID: monitored_entity,
+                    CONF_TYPE: CONF_TYPE_COUNT,
+                    CONF_STATE: ["on"],
+                    CONF_END: "{{now()}}",
+                    CONF_START: "{{ today_at() }}",
+                },
+            }
+        )
+        msg = await client.receive_json()
+        assert msg["success"]
+        assert msg["result"] is None
+
+        msg = await client.receive_json()
+        assert msg["event"]["state"] == "2"
+
+
+async def test_options_flow_preview(
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test the options flow preview."""
+    logging.getLogger("sqlalchemy.engine").setLevel(logging.ERROR)
+    client = await hass_ws_client(hass)
+
+    # add state for the tests
+    await hass.config.async_set_time_zone("UTC")
+    utcnow = dt_util.utcnow()
+    start_time = utcnow.replace(hour=0, minute=0, second=0, microsecond=0)
+    t1 = utcnow.replace(hour=3, minute=0, second=0, microsecond=0)
+    t2 = utcnow.replace(hour=4, minute=0, second=0, microsecond=0)
+    t3 = utcnow.replace(hour=5, minute=0, second=0, microsecond=0)
```

Comment:
You can shorten these by omitting the m,s,us fields like you did in the test above
```suggestion
    t1 = utcnow.replace(hour=3)
    t2 = utcnow.replace(hour=4)
    t3 = utcnow.replace(hour=5)
```